engine/src/card.cpp

```diff
@@ -1088,30 +1088,43 @@ void MCCard::timer(MCNameRef mptr, MCParameter *params)
 		MCObject::timer(mptr, params);
 }
 
+bool MCCard::isdeletable()
+{
+    bool t_deletable = true;
+    
+    if (t_deletable)
```

Comment:
I'd suggest structuring this code as follows:

```
  if (!MCObject::isdeletable())
    return false;
if (!getflag(F_C_CANT_DELETE)
  return false;
...
```
i.e. return false immediately when one of things which causes an object not to be deletable returns false

It makes the code clearer and is also more friendly towards branch prediction (if you have an if with no else, it is generally assumed to be unlikely - and not being able to delete an object is unlikely).